lib/Helper/Requirements.php

```diff
@@ -1,20 +1,18 @@
 <?php
-
 namespace OCA\FaceRecognition\Helper;
 
-use OCP\App\IAppManager;
+use OCA\FaceRecognition\Service\ModelService;
 
 class Requirements
 {
-	/** @var \OCP\App\IAppManager **/
-	protected $appManager;
+	/** @var ModelService */
+	protected $modelService;
 
-	/** @var int ID of used model */
-	private $model;
+	public function __construct(ModelService $modelService,
```

Comment:
I am getting following error when visiting `index.php/settings/admin/facerecognition` (I guess Admin.php still calls `Requirements(appManager)` instead of `Requirements(modelService)`:
```
{
   "Exception":"TypeError",
   "Message":"Argument 1 passed to OCA\\FaceRecognition\\Helper\\Requirements::__construct() must be an instance of OCA\\FaceRecognition\\Service\\ModelService, instance of OC\\App\\AppManager given, called in \/home\/branko\/server\/apps\/facerecognition\/lib\/Settings\/Admin.php on line 51",
   "Code":0,
   "Trace":[
      {
         "file":"\/home\/branko\/server\/apps\/facerecognition\/lib\/Settings\/Admin.php",
         "line":51,
         "function":"__construct",
         "class":"OCA\\FaceRecognition\\Helper\\Requirements",
         "type":"->",
         "args":[
            {
               "__class__":"OC\\App\\AppManager"
            },
            1
         ]
      },
      {
         "file":"\/home\/branko\/server\/settings\/Controller\/CommonSettingsTrait.php",
         "line":115,
         "function":"getForm",
         "class":"OCA\\FaceRecognition\\Settings\\Admin",
         "type":"->",
         "args":[

         ]
      },
      {
         "file":"\/home\/branko\/server\/settings\/Controller\/AdminSettingsController.php",
         "line":74,
         "function":"formatSettings",
         "class":"OC\\Settings\\Controller\\AdminSettingsController",
         "type":"->",
         "args":[
            {
               "20":[
                  {
                     "__class__":"OCA\\FaceRecognition\\Settings\\Admin"
                  }
               ]
            }
         ]
      },
      {
         "file":"\/home\/branko\/server\/settings\/Controller\/CommonSettingsTrait.php",
         "line":126,
         "function":"getSettings",
         "class":"OC\\Settings\\Controller\\AdminSettingsController",
         "type":"->",
         "args":[
            "facerecognition"
         ]
      },
      {
         "file":"\/home\/branko\/server\/settings\/Controller\/AdminSettingsController.php",
         "line":65,
         "function":"getIndexResponse",
         "class":"OC\\Settings\\Controller\\AdminSettingsController",
         "type":"->",
         "args":[
            "admin",
            "facerecognition"
         ]
      },
      {
         "file":"\/home\/branko\/server\/lib\/private\/AppFramework\/Http\/Dispatcher.php",
         "line":166,
         "function":"index",
         "class":"OC\\Settings\\Controller\\AdminSettingsController",
         "type":"->",
         "args":[
            "facerecognition"
         ]
      },
      {
         "file":"\/home\/branko\/server\/lib\/private\/AppFramework\/Http\/Dispatcher.php",
         "line":99,
         "function":"executeController",
         "class":"OC\\AppFramework\\Http\\Dispatcher",
         "type":"->",
         "args":[
            {
               "__class__":"OC\\Settings\\Controller\\AdminSettingsController"
            },
            "index"
         ]
      },
      {
         "file":"\/home\/branko\/server\/lib\/private\/AppFramework\/App.php",
         "line":118,
         "function":"dispatch",
         "class":"OC\\AppFramework\\Http\\Dispatcher",
         "type":"->",
         "args":[
            {
               "__class__":"OC\\Settings\\Controller\\AdminSettingsController"
            },
            "index"
         ]
      },
      {
         "file":"\/home\/branko\/server\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php",
         "line":47,
         "function":"main",
         "class":"OC\\AppFramework\\App",
         "type":"::",
         "args":[
            "OC\\Settings\\Controller\\AdminSettingsController",
            "index",
            {
               "__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"
            },
            {
               "section":"facerecognition",
               "_route":"settings.AdminSettings.index"
            }
         ]
      },
      {
         "function":"__invoke",
         "class":"OC\\AppFramework\\Routing\\RouteActionHandler",
         "type":"->",
         "args":[
            {
               "section":"facerecognition",
               "_route":"settings.AdminSettings.index"
            }
         ]
      },
      {
         "file":"\/home\/branko\/server\/lib\/private\/Route\/Router.php",
         "line":297,
         "function":"call_user_func",
         "args":[
            {
               "__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"
            },
            {
               "section":"facerecognition",
               "_route":"settings.AdminSettings.index"
            }
         ]
      },
      {
         "file":"\/home\/branko\/server\/lib\/base.php",
         "line":987,
         "function":"match",
         "class":"OC\\Route\\Router",
         "type":"->",
         "args":[
            "\/settings\/admin\/facerecognition"
         ]
      },
      {
         "file":"\/home\/branko\/server\/index.php",
         "line":42,
         "function":"handleRequest",
         "class":"OC",
         "type":"::",
         "args":[

         ]
      }
   ],
   "File":"\/home\/branko\/server\/apps\/facerecognition\/lib\/Helper\/Requirements.php",
   "Line":11,
   "CustomMessage":"--"
}
```

Reply:
D'Oh!. I was so focused on something else, I never tried it. :sweat_smile:

Reply:
Done.